plugins/flatpak/gs-flatpak.c

```diff
@@ -2048,11 +2049,28 @@ gs_plugin_refine_item_state (GsFlatpak *self,
 	 * unusable and we should show it as updatable in order for the runtime to be
 	 * installed */
 	if (gs_app_is_installed (app) && runtime != NULL && !gs_app_is_installed (runtime)) {
+		GError *local_error = NULL;
+		g_autoptr(GsPluginEvent) event = NULL;
+
 		g_debug ("App '%s' is installed but its runtime '%s' is not; setting the app"
 			 "as updatable for a chance to fix this", gs_app_get_unique_id (app),
 			 gs_app_get_unique_id (runtime));
+
 		gs_app_set_state (app, AS_APP_STATE_UNKNOWN);
 		gs_app_set_state (app, AS_APP_STATE_UPDATABLE_LIVE);
+
+		/* show event if needed */
+		if ((flags & GS_PLUGIN_REFINE_FLAGS_INTERACTIVE) != 0) {
+			event = gs_plugin_event_new ();
+			g_set_error (&local_error, GS_PLUGIN_ERROR, GS_PLUGIN_ERROR_FAILED,
+				     _("The app %s is missing its runtime. "
+				       "Update the app to repair this problem."),
+				     gs_app_get_id (app));
+			gs_flatpak_error_convert (&local_error);
```

Comment:
Is this call necessary when the error is already in the GS error domain?

Reply:
Actually no. C&P I guess. I'll change that.
:+1: